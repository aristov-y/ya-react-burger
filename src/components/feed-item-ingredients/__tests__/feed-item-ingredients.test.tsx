import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import FeedItemIngredients from '../index';
import { useSelector } from 'react-redux';

const store = {
  ingredients: {
    ingredients: [{
      _id: 'main01',
      price: 100,
      name: 'Main 01'
    }, {
      _id: 'main02',
      price: 102,
      name: 'Main 02'
    }, {
      _id: 'sauce01',
      price: 103,
      name: 'Sauce 01'
    }, {
      _id: 'sauce02',
      price: 104,
      name: 'Sauce 02'
    }]
  }
}
const ingredients = ['sauce02', 'sauce02', 'main01', 'main02', 'main01', 'sauce02']

beforeEach(() => {
  (useSelector as jest.Mock).mockImplementation(fn => fn(store))
});

describe('FeedItemIngredients', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<FeedItemIngredients ingredients={ingredients} />);
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
