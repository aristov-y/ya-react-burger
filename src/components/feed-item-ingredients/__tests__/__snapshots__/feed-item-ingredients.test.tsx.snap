// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItemIngredients should render 1`] = `
<div
  className="Ingredients"
>
  <div
    className="Ingredient"
  >
    <div
      className="IngredientImage"
    >
      <img
        alt="Sauce 02"
      />
    </div>
    <div
      className="text_type_main-default"
    >
      Sauce 02
    </div>
    <div
      className="text text_type_digits-default IngredientPrice"
    >
      3
       x 
      104
      CurrencyIcon
    </div>
  </div>
  <div
    className="Ingredient"
  >
    <div
      className="IngredientImage"
    >
      <img
        alt="Main 01"
      />
    </div>
    <div
      className="text_type_main-default"
    >
      Main 01
    </div>
    <div
      className="text text_type_digits-default IngredientPrice"
    >
      2
       x 
      100
      CurrencyIcon
    </div>
  </div>
  <div
    className="Ingredient"
  >
    <div
      className="IngredientImage"
    >
      <img
        alt="Main 02"
      />
    </div>
    <div
      className="text_type_main-default"
    >
      Main 02
    </div>
    <div
      className="text text_type_digits-default IngredientPrice"
    >
      1
       x 
      102
      CurrencyIcon
    </div>
  </div>
</div>
`;
